libraries/stdlib/src/kotlin/annotations/NativeAnnotations.kt

```diff
@@ -46,3 +46,30 @@ public expect annotation class CName(val externName: String = "", val shortName:
 @Retention(AnnotationRetention.BINARY)
 @OptionalExpectation
 expect annotation class FreezingIsDeprecated
+
+/**
+ * This annotation marks the experimental [ObjCName] annotation.
+ */
+@RequiresOptIn
+@Target(AnnotationTarget.ANNOTATION_CLASS)
+@Retention(AnnotationRetention.BINARY)
+@MustBeDocumented
+@OptionalExpectation
```

Comment:
Could you check that it would be possible to use it in common code as an argument of `OptIn` annotations? I remember optional-expectation annotations might be rather restricted in this regard.

Probably it could be just a common annotation, though a special care should be taken in order not to produce `kotlin.native` package in stdlib for jvm, as `native` package name causes problems with JDK tooling.

Reply:
> Is this a big issue? 

I think it would be an issue if this annotation cannot be opted in to the same way as other experimental annotations.
So it would be better to make it just a plain common annotation.
And in order to avoid producing `kotlin.native` package on JVM, it could be placed in the `kotlin.experimental` package, where another experimental annotation already resides:https://github.com/JetBrains/kotlin/blob/f0a75e7843f09c1a69fff7cd1cda21378ebdab89/libraries/stdlib/src/kotlin/experimental/inferenceMarker.kt#L20
If you do that, please also update API dumps for JS and JVM. It could be done by running the following configurations in IDEA:
- Test: public API binary compatibility validator, overwrite results
- Test: stdlib-js public kotlin api test, overwrite results

Reply:
Cool, I have moved `ExperimentalObjCName` to the `kotlin.experimental` package.
FYI running the JVM API dump test also added some API's related to collections to the dump.